lib/response/CallResponse.js

```diff
@@ -9,241 +9,71 @@ var pathSyntax = require("falcor-path-syntax");
 var derefSync = require("./../deref/sync");
 var $ref = require("./../types/ref");
 
-function CallResponse(subscribe) {
-    Observable.call(this, subscribe || subscribeToResponse);
+function CallResponse(model, callPath, args, suffix, paths) {
+    this.callPath = callPath;
+    this.args = args;
+    this.paths = paths;
+    this.suffix = suffix;
+    this.model = model;
+    this._subscribe = subscribeToResponse;
 }
 
 CallResponse.create = ModelResponse.create;
-
-CallResponse.prototype = Object.create(Observable.prototype);
-CallResponse.prototype.constructor = CallResponse;
-
-CallResponse.prototype.invokeSourceRequest = function invokeSourceRequest(model) {
-    return this;
-};
-
-CallResponse.prototype.ensureCollect = function ensureCollect(model) {
-    return this;
-};
-
-CallResponse.prototype.initialize = function initializeResponse() {
-    return this;
-};
-
-function toObservable(x) {
-    return Rx.Observable.defer(function() {
-        return x;
-    });
-}
+CallResponse.prototype.subscribe = ModelResponse.prototype.subscribe;
 
 function subscribeToResponse(observer) {
-
-    var args = this.args;
+    var callPath = pathSyntax.fromPath(this.callPath);
+    var callArgs = this.args;
+    var suffixes = this.suffix;
+    var extraPaths = this.paths;
     var model = this.model;
-
-    var callPath = pathSyntax.fromPath(args[0]);
-    var callArgs = args[1] || [];
-    var suffixes = (args[2] || []).map(pathSyntax.fromPath);
```

Comment:
@sdesai since it was there before, removing it is a non-backwards compatible change. I think we should have it.